app/models/activity.rb

```diff
@@ -37,11 +38,45 @@ def self.activity_types_list
   end
 
   def old_activity?
-    subject.nil?
+    subject_id.nil?
+  end
+
+  def generate_breadcrumbs
+    generate_breadcrumb subject if subject
   end
 
   private
 
+  def generate_breadcrumb(subject)
+    case subject
+    when Protocol
+      breadcrumbs[:protocol] = subject.name
```

Comment:
You can simplify all blocks moving breadcrumbs out from case:
`breadcrumbs[subject.name.underscore.to_sym] = subject.name`
And after this group them by breadcrumb types.

Reply:
After discussion with @aignatov-bio, we will leave as is for now. 
- When we want to save something else than `.name` from specific `subject` we will have "problems" 
- We can't just move out of switch because than we not covering `else` case than